pkg/util/k8s_utils_test.go

```diff
@@ -0,0 +1,264 @@
+			args{
+				v1PV: nil,
+				v1SC: &jivaSC,
+			},
+			"jiva",
+		},
+		{
+			"Both PV and SC absent",
+			args{
+				v1PV: nil,
+				v1SC: nil,
+			},
+			"unknown",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetCasType(tt.args.v1PV, tt.args.v1SC); got != tt.want {
+				t.Errorf("GetCasType() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetCasTypeFromPV(t *testing.T) {
+	inAnnotationPV.Annotations = map[string]string{"openebs.io/cas-type": "cstor"}
+	inLabelsPV.Labels = map[string]string{"openebs.io/cas-type": "cstor"}
+	type args struct {
+		v1PV *corev1.PersistentVolume
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			"From volume attributes",
+			args{
+				v1PV: &cstorPV1,
+			},
+			"cstor",
+		},
+		{
+			"From labels",
+			args{
+				v1PV: &inLabelsPV,
+			},
+			"cstor",
+		},
+		{
+			"From annotations",
+			args{
+				v1PV: &inAnnotationPV,
+			},
+			"cstor",
+		},
+		{
+			"nil pv",
+			args{
+				v1PV: nil,
+			},
+			"unknown",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetCasTypeFromPV(tt.args.v1PV); got != tt.want {
+				t.Errorf("GetCasTypeFromPV() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetCasTypeFromSC(t *testing.T) {
+	type args struct {
+		v1SC *v1.StorageClass
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			"From provisioner",
+			args{v1SC: &cstorSC},
+			"cstor",
+		},
+		{
+			"From parameters",
+			args{v1SC: &jivaSC},
+			"jiva",
+		},
+		{
+			"SC nil",
+			args{v1SC: nil},
+			"unknown",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetCasTypeFromSC(tt.args.v1SC); got != tt.want {
+				t.Errorf("GetCasTypeFromSC() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetReadyContainers(t *testing.T) {
+	type args struct {
+		containers []corev1.ContainerStatus
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			"valid Values",
+			args{containers: []corev1.ContainerStatus{{Ready: true},{Ready: true},{Ready: false}}},
+			"2/3",
+		},
+		{
+			"Invalid Values",
+			args{containers: []corev1.ContainerStatus{}},
+			"0/0",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetReadyContainers(tt.args.containers); got != tt.want {
+				t.Errorf("GetReadyContainers() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetUsedCapacityFromCVR(t *testing.T) {
+	type args struct {
+		cvrList *cstorv1.CStorVolumeReplicaList
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			"Valid values",
+			args{cvrList: &cstorv1.CStorVolumeReplicaList{Items: []cstorv1.CStorVolumeReplica{{Status: cstorv1.CStorVolumeReplicaStatus{
+				Phase:              "Init",
+				Capacity:           cstorv1.CStorVolumeReplicaCapacityDetails{Total: "5.0 GiB", Used: "2.5 GiB"},
```

Comment:
Adding a different value for the CVR which is stuck in init, to show that this used capacity will not be counted
```suggestion
				Capacity:           cstorv1.CStorVolumeReplicaCapacityDetails{Total: "5.0 GiB", Used: "2.1 GiB"},
```
maybe add another healthy CVR with a Healthy status.Phase & a different size to show that the first Healthy CVR's used capacity is taken in this method.